src/Lean/Meta/Basic.lean

```diff
@@ -1900,6 +1900,10 @@ abbrev isDefEqGuarded (t s : Expr) : MetaM Bool :=
 def isDefEqNoConstantApprox (t s : Expr) : MetaM Bool :=
   approxDefEq <| isDefEq t s
 
+/-- Shorthand for `isDefEq (mkMVar mvarId) val` -/
+def _root_.Lean.MVarId.checkedAssign (mvarId : MVarId) (val : Expr) : MetaM Bool :=
```

Comment:
I'd expect this to return `Unit`

Reply:
`throwError` or `panic` if `isDefEq` fails?

Reply:
I am never quite sure what’s the right one to use. I’d lean towards `throwError` because `panic` is confusing if things then continue to run and it may make sense to run a chunk of `MetaM` code and catch all such errors. But one could also argue that then the programmer should use the `Bool` variant? What would you expect?

Reply:
It am also not sure. I used `Bool` because our current use-cases for the `isDefEq (mkMVar mvarId) v` idiom are all testing whether the assignment failed or not. For debugging programming mistakes where the assignment **must** not fail, I am inclined to have another function that returns `Unit`, and panics if the assignment fails. It is a personal preference because I debug panics using `LEAN_ABORT_ON_PANIC=1 lldb stage1/bin/lean`

Reply:
I will merge the PR as-is since it fixes the bug, but let's continue the discussion. Please feel free to submit a PR with APIs that are convenient for you.